### DISCUSSION TITLE
PATH environment variable in AppImage doesn't include JAVA_HOME

### DISCUSSION THREAD

**Discussion by ppazos** [AppImage/AppImageKit#1317] [Q&A]
Hi, I'm packaging an Electron app in AppImage and it uses a CLI app inside, which is executed using the java command, which is located in the path $JAVA_HOME/bin, and that is part of my system's $PATH.

The app works on dev, bit when packaged, the $PATH en car changes, it's not my system's $PATH, trailering in the java program not being found.

I'm stuck there and I can't distribute my app this way (I'm new to electron and AppImage).

How can I add the user's $JAVA_HOME en car so that the AppImage can find the java command installed on any user computer?

Thanks!

**Comment by ppazos** [comment]
No one?

**Reply by ppazos** [reply]
UPDATE: when I run the app from the terminal, the system JAVA_HOME and PATH are recognized, the issue is when I run by double clicking on the AppImage from a folder window.

**Comment by probonopd** [comment]
Hello. You may want to write your own `AppRun` script that exports `JAVA_HOME`.

In case you'd like more specific support, it would help to share your (non/half-working) AppImage so that we can have a closer look.
